raiden/transfer/mediated_transfer/events.py

```diff
@@ -562,6 +564,25 @@ def from_dict(cls, data: Dict[str, Any]) -> 'EventUnlockFailed':
         return restored
 
 
+class EventNewBalanceProofReceived(Event):
+    """ Event for newly received balance proofs. Useful for notifying monitoring services. """
+
+    def __init__(self, balance_proof: BalanceProofSignedState) -> None:
+        self.balance_proof = balance_proof
+
+    def to_dict(self) -> Dict:
+        return {
+            'balance_proof': self.balance_proof.to_dict(),
+        }
+
+    @classmethod
+    def from_dict(
+            cls: Type[T_EventNewBalanceProofReceived],
+            data: Dict,
+    ) -> T_EventNewBalanceProofReceived:
```

Comment:
I am not sure why you had to declare a type for this. The type would be `EventNewBalanceProofReceived`, no need to declare yet another one in typing.

Reply:
please follow the conversation at https://github.com/raiden-network/raiden/pull/3374#discussion_r253424031 and https://github.com/raiden-network/raiden/pull/3374/commits/69831a8fab12ea38c4ab98ae1122792d7a5d28ae
tldr; class is not available inside class definition

Reply:
None of our other events / state changes / states declare this. Otherwise we would have hundreds of types. Please follow the same convention used in other events in the same file.